src/Lean/Server/Requests.lean

```diff
@@ -171,80 +237,70 @@ def withWaitFindSnapAtPos
     (x := f)
 
 open Language.Lean in
-/-- Finds the first `CommandParsedSnapshot` fulfilling `p`, asynchronously. -/
-partial def findCmdParsedSnap (doc : EditableDocument) (p : CommandParsedSnapshot → Bool) :
-    Task (Option CommandParsedSnapshot) := Id.run do
+/-- Finds the first `CommandParsedSnapshot` containing `hoverPos`, asynchronously. -/
+partial def findCmdParsedSnap (doc : EditableDocument) (hoverPos : String.Pos)
+    (config : Language.SnapSelectionConfig) : Task (Option CommandParsedSnapshot) := Id.run do
```

Comment:
What is an example where we do *not* want to include stop/trailing in `findCmdParsedSnap`? Isn't this only relevant when we get to nesting, i.e. in the info tree?

Reply:
I've adjusted the command snapshot selection logic so that it doesn't need a config.